### ISSUE TITLE
New azurerm_subscription resources are not added to azurerm_management_group which has an existing list of subscription_ids

### ISSUE THREAD

**Issue by FofM** [hashicorp/terraform-provider-azurerm#18097]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.0.11

### AzureRM Provider Version

2.99.0

### Affected Resource(s)/Data Source(s)

azurerm_management_group; azurerm_subscription 

### Terraform Configuration Files

```hcl
I have included mainly the relevant definitions, because the whole code is quite big and has 4 different customized modules.

#from a subscription defining module 
resource "azurerm_subscription" "main" {
  for_each = var.subscriptions

  subscription_name = each.key
  billing_scope_id  = data.azurerm_billing_mca_account_scope.main.id
}

resource "azurerm_role_assignment" "subscription" {
  for_each = local.iam_map

  scope                = "/subscriptions/${azurerm_subscription.main[each.value.subscription_name].subscription_id}"
  role_definition_name = each.value.role
  principal_id         = each.value.principal_id
}

output "subscription_ids" {
  value = local.out_map
}

#from a management defining module
resource "azurerm_management_group" "level_1" {
  for_each = local.azurerm_management_group_level_1

  name                       = each.value.name
  display_name               = each.value.display_name
  parent_management_group_id = "${local.provider_path.management_groups}${each.value.parent_management_group_id}"
  subscription_ids           = try(each.value.subscription_ids, null)
}

output "management_group_output" {
  value = local.core_map
}

#from the main terraform that is called for creation of the resources
module "subscription" {
    new_test_subscription = {
      rbac = [
        {
          role         = ["Owner"]
          principal_id = ["aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"]
        }
      ]
    }
}

module "management-group" {
  root_parent_id = "parent_root"
  management_groups = {
    test-group = {
      parent_management_group_id = "parent_root"
      display_name               = "test-group"
      name                       = "aaaaaaaa-bbbb-aaaa-aaaa-aaaaaaaaaaaa"
      subscription_ids = [
        module.subscription.subscription_ids["existing_test_subscription"],
        module.subscription.subscription_ids["new_test_subscription"]
      ]
    }
  }
}
```

### Debug Output/Panic Output

```shell
module.subscription.azurerm_subscription.main["new_test_subscription"]: Creating...
module.subscription.azurerm_subscription.main["new_test_subscription"]: Still creating... [10s elapsed]
module.subscription.azurerm_subscription.main["new_test_subscription"]: Still creating... [20s elapsed]
module.subscription.azurerm_subscription.main["new_test_subscription"]: Still creating... [2m0s elapsed]
module.subscription.azurerm_subscription.main["new_test_subscription"]: Creation complete after 2m1s [id=/providers/Microsoft.Subscription/aliases/db9a818a-28be-46ac-84c1-7ff59dc7efc7]
module.subscription.azurerm_role_assignment.subscription["new_test_subscription_dev-Owner-c89a9a8d-fab9-427e-9ced-7e6412cf2b2a"]: Creating...
module.subscription.azurerm_role_assignment.subscription["new_test_subscription_dev-Owner-c89a9a8d-fab9-427e-9ced-7e6412cf2b2a"]: Creation complete after 22s [id=/subscriptions/aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa/providers/Microsoft.Authorization/roleAssignments/3fb816b4-6047-b54c-a79f-87a0338f4534]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

management_groups = [
  {
    "/providers/Microsoft.Management/managementGroups/aaaaaaaa-bbbb-aaaa-aaaa-aaaaaaaaaaaa" = {
      "display_name" = "test-group"
      "name" = "aaaaaaaa-bbbb-aaaa-aaaa-aaaaaaaaaaaa"
      "parent_management_group_id" = "parent_root"
      "policy" = []
      "rbac" = [
        {
          "group" = []
          "role" = []
        },
      ]
      "subscription_ids" = [
        "c4d9197a-7b15-4742-9d10-d7b87a05aeb3",
        "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
      ]
    }
  },
]
[01G9Y7GTJGCG6X3Y48YZ4SJAJ1] Changes applied successfully
```

### Expected Behaviour

I expect for the new subscription to be created and added to the management group as it is visible in the output.
Basically if there is an azure management group "foo" containing a subscription "bar" and subscription "baz" is created with the above terraform then "baz" should be placed under "foo" so it will look
```
FOO
|--baz
+--bar
```

### Actual Behaviour

Unlike what is visible in the terraform output the `new_test_subscription` is created but it is not added under the management group `test-group`. It is placed in the Tenant Root Group

Basically after applying it looks like this

```
Tenant Root Group
|--new_test_subscription
parent_root
|--Test Group
     |--existing_test_subscription
```

It is only added if I run the terraform apply twice

### Important Factoids

Azure region: North Europe, executed via Spacelift

**Comment by aristosvo** [comment]
To create subscriptions in the right Management Group, I'd like to specify it in the request. As this requires a newer API version, I made[ a request to include a newer version](https://github.com/hashicorp/go-azure-sdk/issues/164)

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against v2.x of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
